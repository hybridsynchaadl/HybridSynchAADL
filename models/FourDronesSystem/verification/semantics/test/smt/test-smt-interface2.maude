set show timing off .

load ../../interpreter-symbolic2.maude

select SMT-CHECKSAT .
red check-sat(r(0) < [3/1] and r(0) > [1/1]) .
red check-sat(r(0) < [3/1] and r(0) > [1/1]) .

red check-sat(r("a.b",0) < [3/1] and r("a.b",0) > [1/1]) .
red check-sat(r("a",0) < [3/1] and r("b",0) > [1/1]) .

red check-sat(r(0) < [3/1] and r(0) > [1/2]) .
red check-sat(r(0) < [3/7] and r(0) > [1/2]) .

red check-sat(r("a.b",0) < [3/1] and r("a.b",0) > [1/2]) .
red check-sat(r("a",0) < [3/7] and r("b",0) > [1/2]) .
red check-sat(r("a",0) < [3/7] and r("a",0) > [1/2]) .
