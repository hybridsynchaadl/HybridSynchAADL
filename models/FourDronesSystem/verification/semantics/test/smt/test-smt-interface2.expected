==========================================
reduce in SMT-CHECKSAT : check-sat(r(0) < [(3/1).Real] and r(0) > [(1/1).Real])
    .
rewrites: 12
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r(0) < [(3/1).Real] and r(0) > [(1/1).Real])
    .
rewrites: 12
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r("a.b", 0) < [(3/1).Real] and r("a.b", 0) >
    [(1/1).Real]) .
rewrites: 12
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r("a", 0) < [(3/1).Real] and r("b", 0) > [(
    1/1).Real]) .
rewrites: 18
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r(0) < [(3/1).Real] and r(0) > [(1/2).Real])
    .
rewrites: 12
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r(0) < [(3/7).Real] and r(0) > [(1/2).Real])
    .
rewrites: 12
result Bool: (false).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r("a.b", 0) < [(3/1).Real] and r("a.b", 0) >
    [(1/2).Real]) .
rewrites: 12
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r("a", 0) < [(3/7).Real] and r("b", 0) > [(
    1/2).Real]) .
rewrites: 18
result Bool: (true).Bool
==========================================
reduce in SMT-CHECKSAT : check-sat(r("a", 0) < [(3/7).Real] and r("a", 0) > [(
    1/2).Real]) .
rewrites: 12
result Bool: (false).Bool
Bye.
